enterprise/api/filters.py

```diff
@@ -33,25 +35,64 @@ def filter_queryset(self, request, queryset, view):
 
 class EnterpriseCustomerUserFilterBackend(filters.BaseFilterBackend):
     """
-    Allow filtering based on user's email or username on enterprise customer user api endpoint.
+    Allow filtering on the enterprise customer user api endpoint.
     """
 
+    def _filter_by_user_attributes(self, request, queryset):
+        """
+        Filter queryset by email or username.
+        """
+        email = request.query_params.get('email', None)
+        username = request.query_params.get('username', None)
+
+        query_params = {}
+
+        if email:
+            query_params.update(email=email)
+        if username:
+            query_params.update(username=username)
+
+        if query_params:
+            users = User.objects.filter(**query_params).values_list('id', flat=True)
+            return queryset.filter(user_id__in=users)
+
+        return queryset
+
+    def _filter_by_enterprise_attributes(self, request, queryset):
+        """
+        Filter queryset by enterprise_customer_uuid or enterprise role.
+        """
+        enterprise_customer_uuid = request.query_params.get('enterprise_customer_uuid', None)
+        role = request.query_params.get('role', None)
+
+        query_params = {}
+
+        if enterprise_customer_uuid:
+            query_params.update(enterprise_customer_id=enterprise_customer_uuid)
+
+        if role:
+            role_assignment_filters = {'role__name': role}
+            if enterprise_customer_uuid:
+                role_assignment_filters.update(enterprise_customer_id=enterprise_customer_uuid)
+
+            users_with_role = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+                **role_assignment_filters
+            ).values_list('user_id', flat=True)
```

Comment:
This could be a really big result set if `enterprise_customer_uuid` is excluded from the filter.  Maybe we should force `enterprise_customer_uuid` to not be null if `role` is provided?

Reply:
good call, I guess it makes sense to throw an error in this case?